stan/math/prim/fun/log10.hpp

```diff
@@ -48,8 +50,10 @@ inline auto log10(const Container& x) {
  * @param x Container
  * @return Log base-10 of each variable in the container.
  */
-template <typename Container,
-          require_container_st<std::is_arithmetic, Container>* = nullptr>
+template <
+    typename Container,
+    require_container_st<std::is_arithmetic, Container>* = nullptr,
+    require_not_nonscalar_prim_or_rev_kernel_expression_t<Container>* = nullptr>
```

Comment:
I don't think the addition of this require is needed.

Reply:
True. Removed.